imod/typing/grid.py

```diff
@@ -268,3 +268,33 @@ def is_spatial_2D(array: xu.UgridDataArray) -> bool:
 @typedispatch
 def is_spatial_2D(_: object) -> bool:
     return False
+
+
+@typedispatch
+def is_equal(array1: xu.UgridDataArray, array2: xu.UgridDataArray)->bool:
+    # TODO: use .equals function of xugrid once new release has been made
+    if not array1.shape == array2.shape:
+        return False
+    if not (array1.values == array2.values).all():
+        return False
+
+    grid_dataset_1 = array1.ugrid.grid.to_dataset()
+    grid_dataset_2 = array2.ugrid.grid.to_dataset()       
+    return grid_dataset_1.equals(grid_dataset_2)
```

Comment:
This function is fine for now, but we can remove these functions when the new xugrid gets released. See also my other comment.

Reply:
indeed, note that there is a #TODO placed at the beginning of the function mentioning that.